src/main/host/descriptor/socket/inet/mod.rs

```diff
@@ -481,4 +500,25 @@ mod export {
         socket.borrow().update_packet_header(&mut packet);
         packet.into_inner();
     }
+
+    /// Decrement the ref count of the `InetSocketWeak` object. The pointer must not be used after
+    /// calling this function.
+    #[no_mangle]
+    pub extern "C" fn inetsocketweak_drop(socket: *const InetSocketWeak) {
```

Comment:
I think it'd make sense for this to take a `mut` pointer, since it's effectively taking ownership (and destroying)

Reply:
Added.